sdk/node/test/receivers.js

```diff
@@ -0,0 +1,127 @@
+
+/* eslint-env mocha */
+
+const chain = require('../dist/index.js')
+const uuid = require('uuid')
+const assert = require('assert')
+const chai = require('chai')
+const chaiAsPromised = require('chai-as-promised')
+
+chai.use(chaiAsPromised)
+const expect = chai.expect
+
+const client = new chain.Client()
+const signer = new chain.HsmSigner()
+
+function createAccount() {
+  return client.mockHsm.keys.create()
+    .then((key) => {
+      signer.addKey(key, client.mockHsm.signerConnection)
+      return client.accounts.create({
+        alias: `account-${uuid.v4()}`,
+        rootXpubs: [key.xpub],
+        quorum: 1
+      })
+    })
+}
+
+function createAsset() {
+  return client.mockHsm.keys.create()
+    .then((key) => {
+      signer.addKey(key, client.mockHsm.signerConnection)
+      return client.assets.create({
+        alias: `asset-${uuid.v4()}`,
+        rootXpubs: [key.xpub],
+        quorum: 1
+      })
+    })
+}
+
+describe('Receiver', () => {
+
+  describe('Single receiver creation', () => {
+    it('succesfully creates receiver by account alias', () => {
+      return createAccount()
+        .then((account) => client.accounts.createReceiver({ accountAlias: account.alias }))
+        .then((resp) => expect(resp.controlProgram).not.to.be.empty)
+    })
+
+    it('succesfully creates receiver by account id', () => {
+      return createAccount()
+        .then((account) => client.accounts.createReceiver({ accountId: account.id }))
+        .then((resp) => expect(resp.controlProgram).not.to.be.empty)
+    })
+
+    it('rejects receiver creation due to missing params', () => {
+      return createAccount()
+        .then((account) => expect(client.accounts.createReceiver({})).to.be.rejectedWith('CH002'))
+    })
+  })
+
+  describe('Batch receiver creation', () => {
+    let batchResponse = {}
+
+    before(() => {
+      return createAccount()
+        .then((account) => client.accounts.createReceiverBatch([
+          { accountId: account.id }, // success
+          {}, // error
+          { accountAlias: account.alias } // success
+        ]))
+        .then((resp) => {batchResponse = resp})
+    })
+
+    it('returns two successes', () => assert.equal(batchResponse.successes[1], null))
+    it('returns one error', () => assert.deepEqual([batchResponse.errors[0], batchResponse.errors[2]], [null, null]))
+  })
+
+  describe('Pay to receiver', () => {
+    let testAccountAlias, testAssetAlias, testAccountReceiver
+
+    before(() => {
+      return createAccount()
+        .then((account) => testAccountAlias = account.alias )
+        .then(() => createAsset())
+        .then((asset) => testAssetAlias = asset.alias )
+        .then(() => client.accounts.createReceiver({ accountAlias: testAccountAlias }))
```

Comment:
`createAccount` and `createAsset` could live pretty well in a `Promise.all` here too if you like that style.